src/rules/noStaticOnlyClassesRule.ts

```diff
@@ -0,0 +1,106 @@
+        ruleName: "no-static-only-classes",
+        description: Lint.Utils.dedent`
+            Disallows classes containing only static members. Classes
+            with non-empty constructors are ignored.`,
+        rationale: Lint.Utils.dedent`
+            Users who come from a Java-style OO language may wrap
+            their utility functions in an extra class, instead of
+            putting them at the top level.`,
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: [true],
+        type: "functionality",
+        typescriptOnly: false,
+    };
+
+    public static FAILURE_STRING = "Classes containing only static members are disallowed.";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        return this.applyWithWalker(new NoStaticOnlyClassesWalker(sourceFile, this.ruleName, this.ruleArguments));
+    }
+}
+
+class NoStaticOnlyClassesWalker extends Lint.AbstractWalker<string[]> {
+    public walk(sourceFile: ts.SourceFile) {
+        for (const statement of sourceFile.statements) {
+            if (isClassDeclaration(statement)
+                && !hasExtendsClause(statement)
+                && !hasImplementsClause(statement)
+                && !isEmptyClass(statement)) {
+                for (const member of statement.members) {
+                    if (!hasStaticModifier(member.modifiers) && !isEmptyConstructor(member)) {
+                        return;
+                    }
+                }
+                if (statement.name !== undefined) {
+                    this.addFailure(statement.name.pos + 1, statement.name.end, Rule.FAILURE_STRING);
+                }
+            }
+        }
+    }
+}
+
+function hasExtendsClause(statement: ts.ClassDeclaration): boolean {
+    return (statement.heritageClauses !== undefined) ? statement.heritageClauses[0].token === ts.SyntaxKind.ExtendsKeyword : false;
+}
+
+function hasImplementsClause(statement: ts.ClassDeclaration): boolean {
+    return (statement.heritageClauses !== undefined) ? statement.heritageClauses[0].token === ts.SyntaxKind.ImplementsKeyword : false;
+}
+
+function hasStaticModifier(modifiers: ts.NodeArray<ts.Modifier> | undefined): boolean {
+    if (modifiers === undefined) {
+        return false;
+    }
+    for (const modifier of modifiers) {
+        if (modifier.kind === ts.SyntaxKind.StaticKeyword) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function isEmptyClass(statement: ts.ClassDeclaration): boolean {
+    const classMembers = statement.members;
+    if (classMembers.length === 0) {
+        return true;
+    } else if (classMembers.length === 1 && classMembers[0].kind === ts.SyntaxKind.Constructor) {
+        return isEmptyConstructor(classMembers[0]);
+    } else {
+        return false;
+    }
+}
+
+function isEmptyConstructor(member: ts.ClassElement): boolean {
+    if (member.kind === ts.SyntaxKind.Constructor
+        && isFunctionWithBody(member)
```

Comment:
this line is not needed because you already check if it has a body on the next line